src/librustdoc/html/markdown.rs

```diff
@@ -380,7 +380,11 @@ impl<'a, 'b, 'ids, I: Iterator<Item = Event<'a>>> Iterator for HeadingLinks<'a,
                     }
                     _ => {}
                 }
-                self.buf.push_back(event);
+                match event {
+                    Event::Start(Tag::Link(_, _, text)) => self.buf.push_back(Event::Text(text)),
```

Comment:
`text` here is the [link title](https://spec.commonmark.org/0.29/#link-title) which shouldn't be shown. We should just ignore it because it's only normally shown when hovering over a link. For example:

```rust
//! # foo [link](https://example.com/ "title")
```

Should simply render as:

![image](https://user-images.githubusercontent.com/7189418/72385122-b1cc8400-3716-11ea-9c7f-dfdefd98eaed.png)

not:

![image](https://user-images.githubusercontent.com/7189418/72384868-336fe200-3716-11ea-95dc-3797b9f5c0cf.png)

As this PR currently does.

Reply:
Gosh, I didn't even know that was a markdown thing.  Well spotted.  I apologise for my over-eager approval.

Reply:
@ollie27 Wo great catch! I misunderstood what it was doing.

@kinnison Don't worry, we all make mistakes. That's how we learn. :)